src/Symfony/Component/Console/Application.php

```diff
@@ -644,6 +648,12 @@ public function find($name)
             $commands = preg_grep('{^'.$expr.'}i', $allCommands);
         }
 
+        // hidden commands can only be matched exactly and shall not appear otherwise
+        $hiddenCommands = $this->removeHiddenCommands($commands);
+        if (isset($hiddenCommands[$name])) {
```

Comment:
this check can be made before the call to removeHiddenCommands, and then removeHiddenCommands can be made to return the filtered $commands:
```php
if (isset($commands[$name])) {
    return $commands[$name];
}
$commands = $this->removeHiddenCommands($commands);
```

Reply:
But as `removeHiddenCommands` currently returns instances, this way we'd need to duplicate the `get()` part (+ probably deflate aliases).

Reply:
Oh, ok!